src/toolbox/shamir_secrets.rs

```diff
@@ -7,14 +8,82 @@ pub struct SecretShare {
     threshold: usize,
 }
 
+#[derive(Clone, Serialize, Deserialize)]
+pub struct Share {
+    pub x: Scalar,              // this is the x in f(x) of our polynomial
+    pub y: Scalar,              // this is the f(x) of our polynomial
+}
+
 impl SecretShare {
+
+    pub fn share(secret: &Scalar, nr_of_shares: usize, threshold: usize) -> SecretShare {
+        let mut coefficients: Vec<Scalar> = Vec::new();
+        let my_secret = secret.clone();
+        coefficients.push(my_secret);  // the coefficient for x^0 is the secret
+        for _ in 1..(threshold) {
+
+            // TODO should this be an independent RNG, or the transcript RNG?
```

Comment:
Take RNG as input, so we don't have to decide it at this spot. Isn't the one from the transcript just a secure random generator? I haven't looked into the details of that one. Anyway, rand::thread_rng() is okay, but better to pass it in.

Reply:
In reading Henry's code, the `transcript_rng` seemed like it wasn't actually random but pseudorandom, and seeded with the transcript itself, so that we got deterministic values out of the RNG.  Since we aren't planning to call `share()` from within our application, I was unclear what calling the `transcript_rng` out-of-sequence might do.  But you're right, passing in the RNG is a better option in any case.  That particular Rust-ism is new to me, so I didn't think of it.

Reply:
Ah, good to know, I should have looked at the details, I should change this at some spots as well in that case...